### PR TITLE
Avoid early exit for bach norm CNL backward kernel to resolve hang issue

### PR DESCRIPTION
**Pull request by xytintel**
Resolve issue: https://github.com/pytorch/pytorch/issues/136007

### CODE REVIEW

**Review by dvrogozh — CHANGES_REQUESTED** [review]
Unfortunately this PR does not resolve issue for me.

**Review by dvrogozh — APPROVED** [review]
Last version of PR fixes the https://github.com/pytorch/pytorch/issues/136007 for me.

To note, commit which introduced early exit for some threads in original cuda kernels is https://github.com/pytorch/pytorch/commit/1207745e98d740c6c275fbf80193cb537f6edf6f. It modified 3 functions:
* `batch_norm_transform_input_channels_last_kernel`
* `batch_norm_backward_reduce_channels_last_kernel`
* `batch_norm_backward_elemt_channels_last_kernel_impl`

Same early exits exist in all 3 kernels ported to XPU, but only `BatchNormBackwardReduceChannelsLastKernelFunctor` has barrier calls as far as I see (@xytintel, please, double check), so no fixes needed in other 2 kernels. Here are links to the early exit calls:

https://github.com/intel/torch-xpu-ops/blob/c931d66f27c413d03685aea0a221bcadf20aa4bf/src/ATen/native/xpu/sycl/BatchNormKernels.cpp#L1913
https://github.com/intel/torch-xpu-ops/blob/c931d66f27c413d03685aea0a221bcadf20aa4bf/src/ATen/native/xpu/sycl/BatchNormKernels.cpp#L1348
https://github.com/intel/torch-xpu-ops/blob/c931d66f27c413d03685aea0a221bcadf20aa4bf/src/ATen/native/xpu/sycl/BatchNormKernels.cpp#L2657

**Comment by xytintel** [comment]
@dvrogozh Double checked, out of the above 3 kernels, only `BatchNormBackwardReduceChannelsLastKernelFunctor` need barrier for reduction.

**Review by dvrogozh — APPROVED** [review]
looks good

**Review by EikanWang — APPROVED** [review]
